HTML-API: Backport updates from Core

dmsnell opened the pull request:
## What?

This is a "blessed" PR, bringing back changes that were introduced to the HTML API in Trac/SVG that need to exist in Gutenberg to avoid introducing breakage and disharmony between the two repositories.

### Notes

 - this patch restores 6.2 files which were erroneously updated with backports from `trunk`. now the `compat/wordpress-6.2` directory mirrors the HTML API as currently found in the `6.2` branch of `WordPress-develop`
 - while all of the HTML API classes were modified in `WordPress-develop/trunk` only the Tag Processor itself has been brought over, because the other files only involve docblock changes, not behavioral changes, and bringing them over also would require renaming everything inside the Tag Processor to use the `Gutenberg_..._6_3`-wrapped versions of those classes.

## Why?

This is part of a normal process, whereby any updates to SVN need to be back-ported into Gutenberg.

## How?

 - Copy updates from WordPress-develop repo into Gutenberg repo
   - from `WordPress-develop/6.2` copy all files into `lib/compat/wordpress-6.2/html-api/`
   - from `WordPress-develop/trunk` copy just the tag processor into `lib/compat/wordpress-6.3/html-api`
 - Manually verify that the `Gutenberg_..._6_3` class name remains in the 6.3 compat layer instead of the `WP_...` name.

## Testing

Since these tests are already reviewed, tested, and merged into Core, they do not need additional review here. Please ensure that the test suite passes and that nothing was overlooked in the backport process, something which could lead to obvious and catastrophic breakage.

cc: @sergeybiryukov (any changes in Trac/SVN need to have a corresponding PR here like this one, which incorporates the changes you applied).

spacedmonkey commented on lib/compat/wordpress-6.3/html-api/class-wp-html-attribute-token.php:
```diff
@@ -0,0 +1,90 @@
+<?php
+/**
```
Coding standard is going to need to be fixed.

dmsnell replied:
cc: @SergeyBiryukov - looks like someone missed this when approving your change.

@spacedmonkey this is code that has already been accepted in Core. can you fix the issue in Core and create a follow-up back-port into Gutenberg?

until the linting tools are fixed this is going to continue to be a recurring problem. let's not introduce breakage between the two halves of Core because our coding standards are incomplete.

dmsnell replied:
nevermind - I realized this is actually just a temporary thing introducing these new files. thanks @spacedmonkey - not going to be a problem

dmsnell committed:
HTML-API: Backport updates from Core

This is a "blessed" PR, bringing back changes that were introduced to the HTML API
in Trac/SVG that need to exist in Gutenberg to avoid introducing breakage and
disharmony between the two repositories.

cc: @sergeybiryukov

SergeyBiryukov reviewed (approved):
Thanks for bringing over these changes from core! Looks good to me :+1:

SergeyBiryukov reviewed (approved):
Thanks for bringing over these changes from core! Looks good to me :+1:

dmsnell commented:
@hellofromtonya looks like we have another defect in the linting here, as it's blocking this PR, which is code already accepted and merged in Core. I'd like to merge this, but I don't like that we have a rejected CI stage.

Seems like this happens almost every time we backport code from Core. Is someone working on addressing these problems?
